internal/controller/controllers/preprovisioningimage_controller.go

```diff
@@ -114,6 +114,17 @@ func (r *PreprovisioningImageReconciler) Reconcile(origCtx context.Context, req
 		return r.AddIronicAgentToInfraEnv(ctx, log, infraEnv)
 	}
 
+	if infraEnv.Status.CreatedTime == nil {
+		log.Info("InfraEnv image has not been created yet. Requeuing and retrying again soon")
+		setNotCreatedCondition(image)
+		err = r.Status().Update(ctx, image)
+		if err != nil {
+			log.WithError(err).Error("failed to update status")
+			return ctrl.Result{}, err
+		}
+		return ctrl.Result{Requeue: true, RequeueAfter: InfraEnvImageCooldownPeriod}, nil
```

Comment:
Have we considered adding a `Watches` on InfraEnv resources instead of this requeue?
An example of how we do it on other places (see NMState): 
https://github.com/openshift/assisted-service/blob/80c73bc2ce02fd4d9ac84d47776f8c49b7d0b47a/internal/controller/controllers/infraenv_controller.go#L725-L731

Reply:
We have a watch on an event channel already. Let me check to see if we'll get an event for this.
I did this mostly because this is how it is already being done for the "cooldown" check below, but I suppose that's a different use case.

Reply:
I suppose using that would still give us a race between the infraenv controller saving a change back to the kubeapi and the preprovisioning image controller reading that infraenv from the kube api after getting an event from the service code, right?

Feels like a direct watch would be better assuming we only don't need to be notified in situations where nothing has actually changed in the CR.

Curious about why we went this direction initially ...

Based on the [original commit](https://github.com/openshift/assisted-service/commit/18fcad2904816ba9023f2fec1550089989d5f531) that added this in place of a direct watch it sounds like we have an underlying issue that coincidentally I've trying to track down separately. It feels like we should be updating the urls in the infraenv CR more frequently than we currently are.

I was doing a test with static networking config and couldn't seem to get the new config to propagate through ironic to the hosts and I think this is why. The URL must not be changing which would not trigger ironic to re-download the initrd/iso.

I'll put a hold on this until I can sort out this issue.

Reply:
I'm working on just adding a subsystem test to ensure this all works as we assume it will.

I'm going to try to remove this requeue, but not change the current watches and make sure the new test reproduces the issue I saw originally and also passes with these changes (adds the new condition and then sets the image when the infraenv is fixed).

Reply:
Unsure I understand the issue here, the preprovisioning image controller is watching infraEnv:
https://github.com/openshift/assisted-service/blob/3b5982f3d050dae906fef918d5ece9039de58aa8/internal/controller/controllers/preprovisioningimage_controller.go#L256
I think we can remove the requeue